x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -668,16 +687,38 @@ const updateExistingDataStream = async ({
         }),
       { logger }
     );
-    // if update fails, rollover data stream
+
+    // if update fails, rollover data stream and bail out
   } catch (err) {
+    logger.error(`Mappings update for ${dataStreamName} failed`);
+    logger.error(err);
+
     await rolloverDataStream(dataStreamName, esClient);
     return;
   }
+
+  // Trigger a rollover if the index mode or source type has changed
+  if (
+    currentIndexMode !== settings?.index?.mode ||
+    // @ts-expect-error Property 'mode' does not exist on type 'MappingSourceField'
+    currentSourceType !== mappings?._source?.mode
+  ) {
+    logger.debug(
```

Comment:
As this will be pretty uncommon, and just thinking for SDHs like "why have I got so many backing indices" it may be useful to have this at info level to diagnose the problem. Not 100% on this just a thought

Reply:
Yeah this one can probably be `info` - the others might be noisy as we check quite a few data streams on package installation, etc. Will update.